pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -335,7 +334,13 @@ private void searchForCandidateBroker(NamespaceBundle bundle,
 
             if (candidateBroker == null) {
                 if (!this.loadManager.get().isCentralized() || pulsar.getLeaderElectionService().isLeader()) {
-                    candidateBroker = getLeastLoadedFromLoadManager(bundle);
+                    Optional<String> availableBroker = getLeastLoadedFromLoadManager(bundle);
+                    if (!availableBroker.isPresent()) {
```

Comment:
The `lookupFuture` is already of type `Optional<LookupResult>`. Shouldn't be enough to just propagate the empty optional value? (eg: complete successfully with no value).

Reply:
make sense.